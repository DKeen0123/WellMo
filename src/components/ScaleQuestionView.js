import React from 'react';
import Scale from './Scale';
import Question from './Question';
import Submit from './Submit';

const ScaleQuestionView = props => {
  return (
    <div>
      <Question />
      <Scale handleTextBoxInput={props.handleTextBoxInput} />
      <p id="score">{props.score}</p>
      <Submit handleDisplayClick={props.handleDisplayClick} />
    </div>
  );
};

export default ScaleQuestionView;
