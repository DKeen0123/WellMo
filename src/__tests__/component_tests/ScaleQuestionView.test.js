import React from 'react';
import { shallow, mount } from 'enzyme';
import ScaleQuestionView from '../../components/ScaleQuestionView';

describe('ScaleQuestionView', () => {
  let scaleView = shallow(<ScaleQuestionView />);

  describe('Rendering components', () => {
    it('renders correctly', () => {
      expect(scaleView).toMatchSnapshot();
    });

    describe('renders all components', () => {
      it('renders a scale component', () => {
        expect(scaleView.find('Scale').exists()).toBe(true);
      });

      it('renders a Question component', () => {
        expect(scaleView.find('Question').exists()).toBe(true);
      });

      it('renders an Submit component', () => {
        expect(scaleView.find('Submit').exists()).toBe(true);
      });
      
    });
  });
});
