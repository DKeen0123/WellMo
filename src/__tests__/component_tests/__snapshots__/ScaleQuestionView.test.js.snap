// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleQuestionView Rendering components renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScaleQuestionView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Question />,
        <Scale
          handleTextBoxInput={undefined}
/>,
        <p />,
        <Submit
          handleDisplayClick={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleTextBoxInput": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleDisplayClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Question />,
          <Scale
            handleTextBoxInput={undefined}
/>,
          <p />,
          <Submit
            handleDisplayClick={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleTextBoxInput": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleDisplayClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
